bzlib version compile error on windows

Not sure if this is a bug in bzlib - bzlibVersion() now gives a compile error on windows:
```
D:/a/spatial-model-editor/spatial-model-editor/gui/dialogs/dialogabout.cpp: In constructor 'DialogAbout::DialogAbout(QWidget*)':
D:/a/spatial-model-editor/spatial-model-editor/gui/dialogs/dialogabout.cpp:106:24: error: expected primary-expression before '__attribute__'
  106 |                        BZ_API(BZ2_bzlibVersion)()));
      |                        ^~~~~~
make[2]: *** [gui/CMakeFiles/gui.dir/build.make:182: gui/CMakeFiles/gui.dir/dialogs/dialogabout.cpp.obj] Error 1
make[2]: Leaving directory '/d/a/spatial-model-editor/spatial-model-editor/build'
```
Will just remove this from the about dialog box for now

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -97,7 +97,7 @@ jobs:
       - name: Cache ccache
         uses: actions/cache@v3
         with:
-          path: D:\a\_temp\msys64\home\runneradmin\ccache
+          path: C:\Users\runneradmin\AppData\Local\ccache
           key: ccache-${{ github.job }}-${{ runner.os }}-${{ github.sha }}
           restore-keys: ccache-${{ github.job }}-${{ runner.os }}-
       - name: Get static libs

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -114,7 +114,7 @@ jobs:
       - name: Cache ccache
         uses: actions/cache@v3
         with:
-          path: D:\a\_temp\msys64\home\runneradmin\ccache
+          path: C:\Users\runneradmin\AppData\Local\ccache
           key: ccache-${{ github.job }}-${{ runner.os }}-${{ github.sha }}
           restore-keys: ccache-${{ github.job }}-${{ runner.os }}-
       - name: Download static libraries

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -1,7 +1,7 @@
 repos:
 
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.3.0
+    rev: v4.4.0
     hooks:
     - id: check-yaml
     - id: end-of-file-fixer
@@ -10,7 +10,7 @@ repos:
     - id: requirements-txt-fixer
 
   - repo: https://github.com/psf/black
-    rev: 22.10.0
+    rev: 22.12.0
     hooks:
     - id: black
     - id: black-jupyter
@@ -27,7 +27,7 @@ repos:
     - id: nbstripout
 
   - repo: https://github.com/pre-commit/mirrors-clang-format
-    rev: v14.0.6
+    rev: v15.0.4
     hooks:
       - id: clang-format
 
@@ -37,7 +37,7 @@ repos:
       - id: beautysh
 
   - repo: https://github.com/python-jsonschema/check-jsonschema
-    rev: 0.18.3
+    rev: 0.19.2
     hooks:
       - id: check-github-workflows
       - id: check-readthedocs

--- a/ci/windows-gui.sh
+++ b/ci/windows-gui.sh
@@ -43,7 +43,7 @@ cmake .. \
     -DSME_EXTRA_CORE_DEFS=$SME_EXTRA_CORE_DEFS
 make -j2 VERBOSE=1
 
-ccache -s
+ccache -s -v
 
 # check dependencies
 objdump.exe -x sme/sme.cp310-win_amd64.pyd > sme_obj.txt

--- a/gui/dialogs/dialogabout.cpp
+++ b/gui/dialogs/dialogabout.cpp
@@ -102,10 +102,8 @@ DialogAbout::DialogAbout(QWidget *parent)
                        CEREAL_VERSION_PATCH));
   libraries.append(dep("zlib", "https://zlib.net/", ZLIB_VER_MAJOR,
                        ZLIB_VER_MINOR, ZLIB_VER_REVISION));
-  // Should use BZ_API(BZ2_bzlibVersion)() here for version number
-  // but currently this gives a compile error on windows, see
-  // https://github.com/spatial-model-editor/spatial-model-editor/issues/843
-  libraries.append(dep("bzip2", "https://www.sourceware.org/bzip2", "1.0.8"));
+  libraries.append(
+      dep("bzip2", "https://www.sourceware.org/bzip2", BZ2_bzlibVersion()));
   libraries.append(dep("pagmo", "https://esa.github.io/pagmo2",
                        PAGMO_VERSION_MAJOR, PAGMO_VERSION_MINOR,
                        PAGMO_VERSION_PATCH));

